### ISSUE TITLE
All components with events should follow the controlled component pattern

### ISSUE THREAD

**Issue by moog16** [material-components/material-components-web-react#785]
Similar to text-field, all components with events should follow the [controlled component pattern](https://reactjs.org/docs/forms.html#controlled-components). This gives the developer more control over the state of the component. This should also simplify some of the component's logic surrounding events.

- [ ] checkbox
- [ ] dialog
- [ ] chips
- [ ] IconButton
- [ ] drawer

related #772

**Comment by lucasecdb** [comment]
drawer too, right?

**Comment by moog16** [comment]
@lucascdb drawer too yes!

**Comment by moog16** [comment]
Closing for

**Comment by moog16** [comment]
@lucasecdb drawer actually follows the controlled component pattern.

**Comment by lucasecdb** [comment]
it has the API for a controlled component, but it doesn't "respect" it. like, if I don't do anything on the `onClose` callback, the drawer will close anyway, even though the `open` prop is still set to `true`.

**Comment by moog16** [comment]
You are right...I checked dismissible, but only removed the `open` prop. Adding it back. Thanks

**Comment by Jaredk3nt** [comment]
Are there plans to support the controlled components pattern for `Menu` as well?
